src/lib-components/TabList.vue

```diff
@@ -0,0 +1,144 @@
+<script setup>
+import { computed, defineAsyncComponent, provide, ref, useSlots } from 'vue'
+import { useTheme } from '@/composables/useTheme'
+
+const { alignment } = defineProps({
+  alignment: {
+    type: String,
+    default: 'left',
+  },
+})
+
+const SvgIconCalendar = defineAsyncComponent(() =>
+  import('ucla-library-design-tokens/assets/svgs/icon-calendar.svg')
+)
+
+const SvgIconList = defineAsyncComponent(() =>
+  import('ucla-library-design-tokens/assets/svgs/icon-list.svg')
+)
+
+const slots = useSlots()?.default?.()
+
+const tabItems = ref(slots.map((tabItem) => {
+  return tabItem.props
+}))
+
+const activeTab = ref(tabItems.value[0].title)
+
+provide('activeTab', activeTab)
+
+const iconMapping = {
+  'icon-calendar': {
+    icon: SvgIconCalendar,
+    label: 'Calendar'
+  },
+
+  'icon-list': {
+    icon: SvgIconList,
+    label: 'List'
+  },
+}
+
+// Computed
+const parsedAriaLabel = computed(() => {
+  const tabTitle = hyphenateTabName(activeTab.value)
+  return `panel-${tabTitle}`
+})
+
+const theme = useTheme()
+
+const classes = computed(() => {
+  return ['tab-list', theme?.value || '']
+})
+
+// Methods
+function setTabId(tabName) {
+  const tabTitle = hyphenateTabName(tabName)
+  return `tab-${tabTitle}`
+}
+
+function setTabAriaControl(tabName) {
+  const tabTitle = hyphenateTabName(tabName)
+  return `panel-${tabTitle}`
+}
+
+function switchTab(tabName) {
+  activeTab.value = tabName
+
+  const tabId = setTabId(tabName)
+  document.getElementById(tabId).focus()
+}
+
+function hyphenateTabName(str) {
+  return str.toLowerCase().replaceAll(' ', '-')
+}
+
+function keydownHandler(e) {
+  const tabTitleList = tabItems.value.map(obj => obj.title)
+
+  const activeIndex = tabTitleList.indexOf(activeTab.value)
+
+  let targetTab
+
+  switch (e.key) {
+    case 'ArrowLeft':
+      if (activeIndex - 1 < 0)
+        targetTab = tabTitleList[tabTitleList.length - 1]
+      else
+        targetTab = tabTitleList[activeIndex - 1]
+
+      switchTab(targetTab)
+      break
+    case 'ArrowRight':
+      if (activeIndex + 1 > tabTitleList.length - 1)
+        targetTab = tabTitleList[0]
+      else
+        targetTab = tabTitleList[activeIndex + 1]
+
+      switchTab(targetTab)
+      break
+    default:
+  }
+}
+</script>
+
+<template>
+  <div :class="[classes, alignment]" role="tabs">
+    <!-- Slot: Dropdown Filters -->
+    <div v-if="$slots.filters" class="filters">
+      <slot name="filters" />
+    </div>
+
+    <div class="tab-list-header" role="tablist">
+      <button
+        v-for="tab in tabItems"
```

Comment:
use vue Refs here, instead of directly modifying the dom element to set the focus
<button
  v-for="(tab, index) in tabItems"
  :ref="(el) => tabRefs[index] = el"

somewhere in the script setup add 
const tabRefs = ref([])